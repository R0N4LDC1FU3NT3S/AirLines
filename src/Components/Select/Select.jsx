import PropTypes from "prop-types"

import "./Select.scss"

const SelectField = ({ name, onChange, onBlur, defaultValue = " -- select an option -- " }) => {
	const optionList = []

	for (let i = 18; i <= 100; i++) {
		optionList.push({ value: i, label: i })
	}

	return (
		<div className="select">
			<select name={name} onChange={onChange} onBlur={onBlur} defaultValue={""}>
				<option value="">{defaultValue}</option>
				{optionList.map(({ value, label }) => (
					<option key={value} value={value}>
						{label}
					</option>
				))}
			</select>
		</div>
	)
}

export default SelectField
