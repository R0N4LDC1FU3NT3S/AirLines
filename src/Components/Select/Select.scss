select {
	/* Reset Select */
	appearance: none;
	outline: 0;
	border: 1px solid gray;
	box-shadow: none;
	/* Personalize */
	flex: 1;
	padding: 0 1em;
	color: black;
	background-color: white;
	background-image: none;
	cursor: pointer;
}
/* Remove IE arrow */
select::-ms-expand {
	display: none;
}
/* Custom Select wrapper */
.select {
	position: relative;
	display: flex;
	width: 100%;
	height: 3em;
	border-radius: 0.25em;
	overflow: hidden;
}
/* Arrow */
.select::after {
	content: "\25BC";
	position: absolute;
	top: 0;
	right: 0;
	padding: 1em;
	background-color: white;
	transition: 0.25s all ease;
	pointer-events: none;
}
/* Transition */
.select:hover::after {
	color: #f39c12;
}
