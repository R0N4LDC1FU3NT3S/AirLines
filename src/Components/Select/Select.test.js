import { shallow } from "enzyme"
import Select from "./Select"

describe("<Select />", () => {
	let wrapper

	beforeEach(() => (wrapper = shallow(<Select />)))

	it("#0. Snapshot", () => {
		expect(wrapper).toMatchSnapshot()
	})

	it("#1. Render correctly", () => {
		expect(wrapper.find("div").length).toEqual(1)
	})
})
