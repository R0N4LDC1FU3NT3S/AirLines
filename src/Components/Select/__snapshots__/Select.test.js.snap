// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> #0. Snapshot 1`] = `ShallowWrapper {}`;
